packages/@ember/-internals/metal/tests/tracked/validation_test.js

```diff
@@ -402,5 +402,25 @@ moduleFor(
         });
       }, /You attempted to update `foo` on `EmberObject`, but it had already been used previously in the same computation/);
     }
+
+    ['@test get() does not entangle in the autotracking stack until after retrieving the value'](
```

Comment:
Can you explain this test? How did it fail before? How does it confirm what it says it is confirming?

Reply:
Before, `get` would consume the tag for the property immediately, then actually get the _value_ of the property. If the property signaled a change via `notifyPropertyChange` in this time, it would throw the backtracking assertion. This is the fundamental problem that led to the bug in Ember Data.

After, we consume all tags for the first time _after_ getting resolved values, whatever they end up being, which means it is safe to signal property changes while calculating the value.